src/stores/CustomRoomTagStore.js

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {Store} from 'flux/utils';
+import dis from '../dispatcher';
+import RoomListStore from './RoomListStore';
+const STANDARD_TAGS_REGEX = /^(m\.(favourite|lowpriority|server_notice)|im\.vector\.fake\.(invite|recent|direct|archived))$/;
+
+function commonPrefix(a, b) {
+    const len = Math.min(a.length, b.length);
+    let prefix;
+    for (let i = 0; i < len; ++i) {
+        if (a.charAt(i) !== b.charAt(i)) {
+            prefix = a.substr(0, i);
+            break;
+        }
+    }
+    if (prefix === undefined) {
+        prefix = a.substr(0, len);
+    }
+    const spaceIdx = prefix.indexOf(' ');
+    if (spaceIdx !== -1) {
+        prefix = prefix.substr(0, spaceIdx + 1);
+    }
+    if (prefix.length >= 2) {
+        return prefix;
+    }
+    return "";
+}
+/**
+ * A class for storing application state for ordering tags in the TagPanel.
+ */
+class CustomRoomTagStore extends Store {
+    constructor() {
+        super(dis);
+
+        // Initialise state
+        this._state = Object.assign({}, {tags: this._getUpdatedTags()});
+
+        this._roomListStoreToken = RoomListStore.addListener(() => {
+            // UGLY: FluxStore doens't emit changes that
```

Comment:
I wonder if there is a better way to create a store that derives it updates from another store like this?

Reply:
When I wanted to do this before I made a thing called an 'Observer' which is entirely a thing I made up, but served as a thing that doesn't store data itself but aggregates updates from other stores (ActiveRoomObserver.js).

Reply:
I went for an "ad-hoc" store in the end, based off `GroupStore`, as a store seems to be closer to what I'm doing than an observer. :+1::-1:?

Reply:
yep, wfm 👍